torch_xla/csrc/xla_graph_executor.cpp

```diff
@@ -613,17 +621,41 @@ XLAGraphExecutor::ExecuteComputationWithBarrier(
   std::shared_ptr<XLAGraphExecutor::Async> async = std::make_shared<Async>(
       &coll, std::move(arguments), placeholders, std::move(cachedComputation));
 
-  auto syncfn = [async, hash]() {
+  auto syncfn = [async, hash, sharding_specs]() {
```

Comment:
why `sharding_specs` needs to be passed from original function? It is init as default value right? We might as well just init it within this function.

Reply:
ok, I guess the point here is we need to supply a proper sharding spec for the output, but it is currently not supported yet. As a follow up, what's the right way to determine the sharding spec for output?

Reply:
Good question, I'm not sure the right way to determine sharding spec for output. @yeounoh might be able to answer or provide more context

Reply:
That's fine, we will leave that as a follow up.